packages/start/src/config/index.ts

```diff
@@ -530,3 +555,66 @@ function tsrRoutesManifest(opts: {
     },
   }
 }
+
+const FILE_EXTENSIONS = ['js', 'jsx', 'ts', 'tsx']
+const HTTP_METHODS = ['GET', 'POST', 'PUT', 'DELETE', 'PATCH']
+
+class TanStackStartServerFileSystemRouter extends BaseFileSystemRouter {
+  toPath(src: string): string {
+    const path = cleanPath(src, this.config)
+
+    const segments = startAPIRouteSegmentsFromPath(path)
+
+    const pathname = segments
+      .map((part) => {
+        if (part.type === 'splat') {
+          return `*splat`
+        }
+
+        if (part.type === 'param') {
+          return `:${part.value}?`
+        }
+
+        return part.value
+      })
+      .join('/')
+
+    return pathname.length > 0 ? `/${pathname}` : '/'
+  }
+
+  toRoute(src: string) {
+    const path = this.toPath(src)
+
+    const [_, exports] = analyzeModule(src)
+
+    const hasAPIExports = !!exports.find((exp) => HTTP_METHODS.includes(exp.n))
+
+    if (!hasAPIExports) {
+      return
+    }
+
+    const exportedAPIHandlers = exports
+      .filter((exp) => HTTP_METHODS.includes(exp.n))
+      .reduce(
+        (handlers, exp) => {
+          handlers[`$${exp.n}`] = {
+            src,
+            pick: [exp.n],
+          }
+          return handlers
+        },
+        {} as Record<string, any>,
+      )
+
+    if (Object.keys(exportedAPIHandlers).length === 0) {
```

Comment:
This can get nuked. We wouldn't get here, if expected identifiers weren't present.